frontend/src/components/shared/FullScreenWrapper/FullScreenWrapper.tsx

```diff
@@ -0,0 +1,140 @@
+import "./FullScreenWrapper.scss"
+
+export type Size = {
+  height?: number
+  width: number
+}
+
+/*
+ * Function responsible for rendering children.
+ * This function should implement the following signature:
+ * ({ height, width }) => PropTypes.element
+ */
+interface Props {
+  children: (props: Size) => React.ReactNode
+  width: number
+}
+
+interface State {
+  expanded: boolean
+  fullwidth: number
+  fullheight: number
+}
+
+/*
+ * A component that draws a button on the top right of the
+ * wrapper element. OnClick, change the element container
+ * to fixed and cover all screen, updating wrapped element height and width
+ */
+class FullScreenWrapper extends PureComponent<Props, State> {
+  static isFullScreen = false
+
+  public state: State = { expanded: false, fullwidth: 0, fullheight: 0 }
+
+  componentDidMount() {
+    this.updateWindowDimensions()
+    window.addEventListener("resize", this.updateWindowDimensions)
+    document.addEventListener("keydown", this.controlKeys, false)
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener("resize", this.updateWindowDimensions)
+    document.removeEventListener("keydown", this.controlKeys, false)
+  }
+
+  controlKeys = (event: any) => {
+    const { expanded } = this.state
+
+    if (event.keyCode === 27 && expanded) {
+      // Exit fullscreen
+      this.zoomOut()
+    }
+  }
+
+  zoomIn = () => {
+    document.body.style.overflow = "hidden"
+    FullScreenWrapper.isFullScreen = true
+    this.setState({ expanded: true })
+  }
+
+  zoomOut = () => {
+    document.body.style.overflow = "unset"
+    FullScreenWrapper.isFullScreen = false
+    this.setState({ expanded: false })
+  }
+
+  convertScssRemValueToPixels = (scssValue: string) => {
+    const remValue = parseFloat(scssValue)
+    return (
+      remValue *
+      parseFloat(getComputedStyle(document.documentElement).fontSize)
+    )
+  }
+
+  updateWindowDimensions = () => {
+    const padding = this.convertScssRemValueToPixels(
+      SCSS_VARS["$fullscreen-padding"]
+    )
+    const paddingTop = this.convertScssRemValueToPixels(
+      SCSS_VARS["$fullscreen-padding-top"]
+    )
+    this.setState({
+      fullwidth: window.innerWidth - padding * 2, // Left and right
+      fullheight: window.innerHeight - (padding + paddingTop), // Bottom and Top
+    })
+  }
+
+  public render(): JSX.Element {
+    const { expanded, fullwidth, fullheight } = this.state
+    const { children, width } = this.props
+
+    let buttonClassName = "overlayBtn stButton-enter"
+    let buttonImage = "fullscreen-enter"
+    let buttonOnClick = this.zoomIn
+    let buttonTitle = "View fullscreen"
+
+    if (expanded) {
+      buttonClassName = "overlayBtn stButton-exit"
+      buttonImage = "fullscreen-exit"
+      buttonOnClick = this.zoomOut
+      buttonTitle = "Exit fullscreen"
+    }
+
+    return (
+      <div className={`stFullScreenFrame${expanded ? "--expanded" : ""}`}>
+        <button
```

Comment:
Is is possible to use a Reactstrap `<Button>` component? See the main menu and toolbar button for examples.

Reply:
Yes I can, but I based on this button: https://github.com/dcaminos/streamlit/blob/58bd06cee4de0b403dbfff55de8a07aea79dbf4a/frontend/src/components/elements/CodeBlock/CopyButton.tsx#L46

We should create a "OverlayButton" component for both cases (and use Reactstrap), but I don't wanna do it on this PR (it's too big right now)

Reply:
Sounds good. Can you create an issue for this?